src/internal/utils.ts

```diff
@@ -315,22 +315,29 @@ export function getName<U extends AnyParamConstructor<any>>(cl: U) {
   const ctor: any = getRightTarget(cl);
   const options: IModelOptions = Reflect.getMetadata(DecoratorKeys.ModelOptions, ctor) ?? {};
   const baseName: string = ctor.name;
+  const customName = options.options?.customName;
 
   if (options.options?.automaticName) {
-    const suffix = options.options?.customName ?? options.schemaOptions?.collection;
+    const suffix = customName ?? options.schemaOptions?.collection;
 
     return !isNullOrUndefined(suffix) ? `${baseName}_${suffix}` : baseName;
   }
 
-  if (typeof options.options?.customName === 'string') {
-    if (options.options.customName.length <= 0) {
+  if (typeof customName === 'string') {
+    if (customName.length <= 0) {
       throw new TypeError(`"customName" must be a string AND at least one character ("${baseName}") [E015]`);
     }
+  }
+
+  if (customName !== undefined && typeof customName !== 'string') {
+    return customName(options);
+  }
 
-    return options.options.customName;
+  if (customName === undefined) {
```

Comment:
```suggestion
  if (isNullOrUndefined(customName)) {
```

Reply:
Done! Thanks for the tips. Like I said, learning experience for me. 

Scott